plan/logical_plan_builder.go

```diff
@@ -2403,3 +2404,30 @@ func getInnerFromParentheses(expr ast.ExprNode) ast.ExprNode {
 	}
 	return expr
 }
+
+func extractTableListForUpdate(refTables []*ast.TableName, sets []*ast.Assignment) []*ast.TableName {
+	// if there's only one tbl , we won't bother
+	if len(refTables) == 1 {
+		return refTables
+	}
+	var tbls []*ast.TableName
+	var tblsMap map[string]*ast.TableName
+
+	for _, a := range sets {
+		c := a.Column
+
+		if c.Table.L != "" {
+			for _, tbl := range refTables {
+				if strings.EqualFold(tbl.Name.L, c.Name.L) {
+					tblsMap[fmt.Sprintf("`%s`.`%s`", tbl.Schema.L, tbl.Name.L)] = tbl
+				}
+			}
+		}
+	}
+
+	for _, tbl := range tblsMap {
```

Comment:
I was wondering why not append it directly in the line 2422?

Reply:
If we got sql: 
```
update ap.record t1 join tp.record t2 on t1.id=t2.id set t1.name='' and t1.email='' 
```
we will get 2 `ast.Assignment`s t1.name and t1.email,if i just append in 2422, it will get duplicated.